dotnet/src/Agents/OpenAI/OpenAIServiceConfiguration.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Net.Http;
+using Azure.Core;
+
+namespace Microsoft.SemanticKernel.Agents.OpenAI;
+
+/// <summary>
+/// Configuration to target a specific Open AI service.
+/// </summary>
+public sealed class OpenAIServiceConfiguration
+{
+    internal enum OpenAIServiceType
+    {
+        AzureOpenAI,
+        OpenAI,
+    }
+
+    /// <summary>
+    /// Produce a <see cref="OpenAIServiceConfiguration"/> that targets an Azure OpenAI endpoint using an API key.
+    /// </summary>
+    /// <param name="apiKey">The API key</param>
+    /// <param name="endpoint">The service endpoint</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    public static OpenAIServiceConfiguration ForAzureOpenAI(string apiKey, Uri endpoint, HttpClient? httpClient = null)
+    {
+        Verify.NotNullOrWhiteSpace(apiKey, nameof(apiKey));
+        Verify.NotNull(endpoint, nameof(endpoint));
+
+        return
+            new()
+            {
+                ApiKey = apiKey,
+                Endpoint = endpoint,
+                HttpClient = httpClient,
+                Type = OpenAIServiceType.AzureOpenAI,
+            };
+    }
+
+    /// <summary>
+    /// Produce a <see cref="OpenAIServiceConfiguration"/> that targets an Azure OpenAI endpoint using an token credentials.
+    /// </summary>
+    /// <param name="credential">The credentials</param>
+    /// <param name="endpoint">The service endpoint</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    public static OpenAIServiceConfiguration ForAzureOpenAI(TokenCredential credential, Uri endpoint, HttpClient? httpClient = null)
+    {
+        Verify.NotNull(credential, nameof(credential));
+        Verify.NotNull(endpoint, nameof(endpoint));
+
+        return
+            new()
+            {
+                Credential = credential,
+                Endpoint = endpoint,
+                HttpClient = httpClient,
+                Type = OpenAIServiceType.AzureOpenAI,
+            };
+    }
+
+    /// <summary>
+    /// Produce a <see cref="OpenAIServiceConfiguration"/> that targets OpenAI services using an API key.
+    /// </summary>
+    /// <param name="apiKey">The API key</param>
+    /// <param name="endpoint">An optional endpoint</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    public static OpenAIServiceConfiguration ForOpenAI(string? apiKey, Uri? endpoint = null, HttpClient? httpClient = null)
```

Comment:
Strongly suggest not adding the BreakingGlass scenario for this extension, as the customer will be better off doing the AzureOpenAI client himself instead of using this.

If we proceed in having a `HttpClient` option then I suggest removing completely the `HttpClient?` property from it, as this is an `IDisposable` object and the caller already has the `httpClient` reference on his side.

Reply:
Updated according to our discussion (and renamed class - `OpenAIClientProvider`).  One thing that surfaced when making the changes was the reliance on the agent to know something about the configuration to produce the channel keys.  This is managed with internal scope and is a critical consideration for not passing teh `OpenAIClient` directly to the assistant-agent.  Please ping me if there are any questions on the channel / channel-key stuff.